### PR TITLE
Update grid topics for date object usage

### LINKED ISSUE #377: Add Note for Grid filtering about string objects when using filtering and sorting
Only date objects should be passed when filter/sort Date type columns. Using of string objects is not allowed.

### PR DESCRIPTION
**Pull request by zdrawku**
Add explanation that additional changes should be made, on application level, for features like sorting and filtering, when it comes to using of `string` values on `Date` column.

Close #377

### CODE REVIEW

File: en/components/grid_filtering.md [hunk 1]
```diff
@@ -26,6 +26,9 @@ There's a default filtering strategy provided out of the box, as well as all the
 <igx-column field="Price" filterable="true" dataType="number"></igx-column>
 ```
 
+> [!NOTE]
+> If values of type `string` are used by column of dataType `Date`, the grid won't parse it to `Date` objects and using of filtering conditions won't be possible. If you want to use `string` objects, additional logic should be implemented on application level, in order to parse the values to `Date` object.
```
Comment (ig-ealbert) [inline en/components/grid_filtering.md]: ... won't parse it to `Date` objects and using filtering conditions...
or
... won't parse it to `Date` objects and usage of filtering conditions...
I like the first one better, it sounds more natural.

Also, 
... should be implemented on the application level
or
... should be implemented on an application level

File: en/components/grid_sorting.md [hunk 2]
```diff
@@ -68,6 +68,10 @@ public ngOnInit() {
     ];
 }
 ```
+
+> [!NOTE]
+> If values of type `string` are used by column of dataType `Date`, the grid won't parse it to `Date` objects and using of igxGrid `sorting` won't work as expected. If you want to use `string` objects, additional logic should be implemented on application level, in order to parse the values to `Date` object.
```
Comment (ig-ealbert) [inline en/components/grid_sorting.md]: Same comments here.

**Comment by zdrawku** [comment]
@ig-ealbert both en and jp versions are updated for sorting and filtering subtopics

**Comment by ig-ealbert** [comment]
I can add the translations today.

**Pull request merged**